lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -722,3 +728,28 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+-- | Extracts the raw xprv bytes as @private key <> chain code@, i.e. without
+-- the public key.
+unXPrvStripPub :: XPrv -> ByteString
+unXPrvStripPub = stripPub . unXPrv
+  where
+    -- Converts  prv <> pub <> cc
+    -- To        prv <>        cc
+    stripPub :: ByteString -> ByteString
+    stripPub xprv' = prv <> chainCode
+      where
+        (prv, rest) = BS.splitAt 64 xprv'
+        (_pub, chainCode) = BS.splitAt 32 rest
+
+-- Create a @XPrv@ from a prv <> cc bytestring.
+--
+-- TODO: I'm not quite sure how this works, despite rountrip test passing.
+-- Sanity check.
+xPrvFromStrippedPubXPrv :: ByteString -> Either String XPrv
+xPrvFromStrippedPubXPrv x = xprv
+    $ CC.unEncryptedKey
+    $ CC.encryptedCreateDirectWithTweak x pass
+  where
+    pass :: ByteString
+    pass = ""
```

Comment:
It'd be better to have this as a function argument and pass an empty passphrase at the call-site! Avoid hard-coding things in functions.

Reply:
In that case we'd have:
```haskell
xPrvFromStrippedPubXPrv :: Passphrase "encryption" -> ByteString -> Either ErrXPrvFromStrippedPubXPrv XPrv
```
taking a password, but
```haskell
unXPrvStripPub :: XPrv -> ByteString
```
not taking a password.

But I guess this is fine as the password would already be "embedded" in the `XPrv`.

That said I'm not quite sure these functions should work regarding passwords. The following test fails:

```haskell
prop_strippedPubRoundtrip
    :: Unencrypted (ByronKey 'RootK XPrv)
    -> Passphrase "encryption"
    -> Property
prop_strippedPubRoundtrip (Unencrypted k) pwd = do
    let f = (xPrvFromStrippedPubXPrv pwd) . unXPrvStripPub . (xPrvChangePass empty pwd) . getRawKey
    let raw = unXPrv . getRawKey
    either (error . show) unXPrv (f k) === raw k
  where
    empty = Passphrase ""
```
because the public keys are different (prv <> pub <> cc) 🤔 
```
"|+S2m;Y\r\246Ru\253\227Y\CAN\233o\249E\223\168\141\253\200L\162WkL>x\DEL\201\170\204\232\ENQ5\DC4\231\143E|C\165\175Y\240WUM@\128\151\203j\216\185oc\178\187p\192\192k\135\vq\CAN\168g31\EOT\159\EOT\160\179`\198\196\160\&1Q=\165\188\&7'dm\210/\139t\ESC\152\205C\146\128\132\172\143\255\GS\200\234\239>\DC1\EM\142\212\199L\165Q\186\197\136\151{3\192\ETB\220"
/=
"x+S2m;Y\r\246Ru\253\227Y\CAN\233o\249E\223\168\141\253\200L\162WkL>x_\201\170\204\232\ENQ5\DC4\231\143E|C\165\175Y\240WUM@\128\151\203j\216\185oc\178\187p\192\DC2\243n!)\130e\250\201\231\236\US\208\ETB`\DC2\201e&\187\244\206 U\172A<i\143\a\v\214\ESC\152\205C\146\128\132\172\143\255\GS\200\234\239>\DC1\EM\142\212\199L\165Q\186\197\136\151{3\192\ETB\220"
```

Reply:
unXPrv would give you the **encrypted** key, so you would need to decrypt it first. It's not that the passphrase is embedded, it is that the key is encrypted. 
In principles calling `unXPrvStripPub` on a key that is encrypted should fail. Also, what exactly is the use for "xPrvFromStrippedPubXPrv" ?

Reply:
> Also, what exactly is the use for "xPrvFromStrippedPubXPrv" ?

It will be needed to parse 96-byte key as a CLI argument. I.e `key child` in the future.

For now, it is only used for roundtrips.

We could move it to `Cardano.CLI`. I contemplated doing that.

Reply:
> It will be needed to parse 96-byte key as a CLI argument. I.e key child in the future.

:+1: Ok, then a passphrase makes little sense here, but using `XPrv` feels also misleading because we do actually treat them and manipulate them as unencrypted key. A newtype could capture that nicely... or at the very least a comment / note explaining why the passphrase is forced to be empty. (Still, I'd add an invariant or runtime check that, when trying to xPrvStripXPub, the given key is unencrypted and fail loudly if it is).